pkg/git/RepositoryManager.go

```diff
@@ -46,24 +46,31 @@ type RepositoryManager interface {
 }
 
 type RepositoryManagerImpl struct {
-	logger *zap.SugaredLogger
+	logger  *zap.SugaredLogger
+	gitUtil *GitUtil
 }
 
-func NewRepositoryManagerImpl(logger *zap.SugaredLogger) *RepositoryManagerImpl {
-	return &RepositoryManagerImpl{logger: logger}
+func NewRepositoryManagerImpl(logger *zap.SugaredLogger, gitUtil *GitUtil) *RepositoryManagerImpl {
+	return &RepositoryManagerImpl{logger: logger, gitUtil: gitUtil}
 }
 
-func (impl RepositoryManagerImpl) Add(location string, url string, auth transport.AuthMethod) error {
+func (impl RepositoryManagerImpl) Add(location string, url string, userName, password string) error {
 	err := os.RemoveAll(location)
 	if err != nil {
 		impl.logger.Errorw("error in cleaning checkoutpath", "err", err)
 		return err
 	}
-	_, err = impl.clone(auth, location, url)
+	err = impl.gitUtil.Init(location, url, true)
```

Comment:
will this be called only once? repeated call may return error

Reply:
yes only once for fresh checkout